cloudferrylib/os/compute/nova_compute.py

```diff
@@ -658,7 +658,7 @@ def deploy_instance(self, create_params, conf):
                 self.wait_for_status(new_id, self.get_status, 'active',
                                      timeout=conf.migrate.boot_timeout,
                                      stop_statuses=[ERROR])
-            except exception.TimeoutException:
+            except (exception.TimeoutException, KeyboardInterrupt):
```

Comment:
VM may be in BUILDING or SCHEDULING state at the time when exception occurs, but may later boot successfully, so make sure you don't delete VM which booted successfully and is in ACTIVE state.